trader/trader.go

```diff
@@ -432,12 +468,18 @@ func (t *Trader) update() bool {
 		if e != nil {
 			log.Printf("error in filter index %d: %s\n", i, e)
 			t.deleteAllOffers(false)
-			return false
+			return metrics.UpdateLoopResult{
+				Success:            false,
+				NumPruneOps:        numPruneOps,
+				NumUpdateOpsDelete: numUpdateOpsDelete,
+				NumUpdateOpsUpdate: numUpdateOpsUpdate,
+			}
 		}
 	}
 
-	log.Printf("created %d operations to update existing offers\n", len(ops))
-	if len(ops) > 0 {
+	numUpdateOpsUpdate = len(ops)
```

Comment:
we haven't used `numUpdateOpsDelete` anywhere.

the numUpdateOps is broken down into two subparts: numUpdateOpsUpdate and numUpdateOpsDelete.
i.e. these update ops can contain ops to create, modify, or delete offers.
(on second thought we need to accommodate the "create" case as well so need to add that metric to the UpdateLoopResult and therefore in amplitude as well).

we need a for loop over all the update ops to count them towards either `create`, `update`, or `delete` ops.
you can call that function "countUpdateOps` as an example and have that function independently tested.